### ISSUE TITLE
[bug] HS13A-1 stuck on "LOC"

### ISSUE THREAD

**Issue by tsheaff** [iMicknl/LoctekMotion_IoT#10]
Not sure this is a bug with this library, but I was playing around with connecting my arduino to the standing desk, and now I've noticed when plugging back in my original `HS13A-1` keypad, it's stuck in the `LOC` state. Not 100% sure how to reproduce, but it will not get out of this setting regardless of how I press the buttons. I've tried:

- hold down M for a long time
- hold down up and down for a long time
- powering on and off the control
- powering on and off the control box itself

After all of these it still just says `LOC` and will not move the desk up and down. I'm not sure how to debug whether it's an issue with the keypad itself, or with the box it connects to. Is any state stored in this box? My box model is `CB38M2B(IB)-1`

Has anyone else run into this issue? Any ideas how to resolve? Perhaps there's a calibration issue it's experiencing where it thinks the desk is on uneven surface or otherwise has faulty state, and has shut down motor controls into the `LOC` state? cc @iMicknl @staxDB 🙏

**Comment by iMicknl** [comment]
Which code did you use? 

I would start with the following steps first:

1. Disconnect your Arduino from the motor control box
2. Remove the power cable of your desk
3. Wait 60 seconds
4. Plug in the power cable of your desk
5. Did this solve your problem?

I have exactly the same control box and control panel, however I am using it with ESPHome + Home Assistant at the moment.

By the way, there are some new users figuring out how it works, thus they created a Discord server. Feel free to [join us on Discord](https://discord.gg/C7TNzUZ9Xf).

**Comment by mad2xlc** [comment]
LOC means lock and can be triggered with a 5 seconds press on memory and also released the same way.
It's not a bug it's a feature :)
I haven't attached my desk to an esp but came across this issue and was worried about this bug. But i think everything is fine here.

@tsheaff 
Were you able to solve the problem already?

**Comment by iMicknl** [comment]
Thanks for the context @mad2xlc! Perhaps we should add this to the FAQ as well. 

I will close this issue for now.

**Comment by tsheaff** [comment]
Hey @iMicknl @mad2xlc apologies for the incredibly slow reply 😆 

I have not managed to fix this issue with the above. Holding down the memory for 5 seconds definitely doesn't fix it, or I would've fixed this months ago. I'm thinking of just buying an new controller. Any thoughts on the best place to purchase one?

**Comment by RinkMax** [comment]
Hi @tsheaff,
I am facing the same problem. Have you been able to solve the problem by now?

**Comment by DJHansford** [comment]
In case anyone else stumbles on this, I had the same problem with my Sanodesk controller (HS13A-1), though I've never connected it to an Arduino board. It got stuck on child lock mode and pressing and holding M for 5 seconds didn't work.

I fixed it by pressing and holding the M button while removing the power socket. It beeped at me through this procedure until the residual power in the system died. Just before shutting off it flashed '5-'. Once powered back on it was unlocked. Could just be dumb luck but maybe it'll help someone else.

**Comment by DJHansford** [comment]
I've just a variation of this bug, googled for the fix I found before and found my own comment (thanks previous me!). However, this time it was different, my keypad was stuck in locked mode but when I pressed the `M` button it only beeped once, not continuously. The fix above wouldn't work as a result. Instead, after playing around for a long time, I pressed the `1` and `2` buttons together for a few seconds and it beeped continuously at me, then pressing the `M` caused the same beeping and my fix above worked.
